File: src/Data/Customer.java
package Data;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;

public class Customer {

	public ArrayList<String> customerArray = new ArrayList<>();
	private String customerName;
	private String phoneNum;
	private String deliveryAddress;
	private FileWriter writeTable = new FileWriter("Customer Info.txt", StandardCharsets.UTF_8, true);
	private BufferedWriter bw = new BufferedWriter(writeTable);
	private PrintWriter out = new PrintWriter(bw);
	private String emailAddress;
	private int amt;


	public Customer(String custName, String phoneNum, String address, String email, int amt) throws IOException {
		this.customerName = custName;
		this.phoneNum = phoneNum;
		this.deliveryAddress = address;
		this.emailAddress = email;
		this.amt = amt;
	}

	public void customerInfo() {
		String formatStr = "%-30s %-30s %-30s %-30s %-30s %-25s%n";
		out.print(String.format(formatStr, "Customer Name", "Customer Phone Number", "Delivery Address", "Email Address", "Order Quantity", "Cost"));
		Order order = new Order(getAmt());
		out.print(String.format(formatStr, getCustomerName(), getPhoneNum(), getDeliveryAddress(), getEmailAddress(), getAmt(), order.total()));
		out.close();
	}

	private String getCustomerName() {
		return customerName;
	}

	private String getPhoneNum() {
		return phoneNum;
	}

	private String getDeliveryAddress() {
		return deliveryAddress;
	}

	public String toString() {
		StringBuilder builder = new StringBuilder();
		for (String value : customerArray) {
			builder.append(value).append(" ");
		}

		return "Customer Data:\n " + builder.toString();
	}

	private String getEmailAddress() {
		return emailAddress;
	}

	private int getAmt() {
		return amt;
	}

}

File: src/data/Customer.kt
package data

import java.io.BufferedWriter
import java.io.FileWriter
import java.io.IOException
import java.io.PrintWriter
import java.nio.charset.StandardCharsets
import java.util.*

class Customer @Throws(IOException::class)
constructor(private val customerName: String, private val phoneNum: String, private val deliveryAddress: String, private val emailAddress: String, private val amt: Int) {

    var customerArray = ArrayList<String>()
    private val writeTable = FileWriter("src/resources/Customer Info.txt", StandardCharsets.UTF_8, true)
    private val bw = BufferedWriter(writeTable)
    private val out = PrintWriter(bw)

    fun customerInfo() {
        val formatStr = "%-30s %-30s %-30s %-30s %-30s %-25s%n"
        out.print(String.format(formatStr, "Customer Name", "Customer Phone Number", "Delivery Address", "Email Address", "Order Quantity", "Cost"))
        val order = Order(amt.toLong())
        out.print(String.format(formatStr, customerName, phoneNum, deliveryAddress, emailAddress, amt, order.total()))
        out.close()
    }

    override fun toString(): String {
        val builder = StringBuilder()
        for (value in customerArray) {
            builder.append(value).append(" ")
        }

        return "Customer data:\n $builder"
    }

}
